pallets/pallet-bonded-coins/src/mock.rs

```diff
@@ -11,3 +46,295 @@ pub fn assert_relative_eq(target: Float, expected: Float, epsilon: Float) {
+		type CreateOrigin = EnsureSigned<AccountId>;
+		type Currency = Balances;
+		type Extra = ();
+		type ForceOrigin = EnsureRoot<AccountId>;
+		type Freezer = ();
+		type MetadataDepositBase = ConstU128<0>;
+		type MetadataDepositPerByte = ConstU128<0>;
+		type RemoveItemsLimit = ConstU32<5>;
+		type RuntimeEvent = RuntimeEvent;
+		type StringLimit = StringLimit;
+		type WeightInfo = ();
+	}
+	parameter_types! {
+		pub const CurrencyDeposit: Balance = 500;
+		pub const MaxCurrencies: u32 = 50;
+		pub const CollateralAssetId: u32 = u32::MAX;
+	}
+
+	impl Config for Test {
+		type AssetId = AssetId;
+		type BaseDeposit = ExistentialDeposit;
+		type CollateralCurrency = Assets;
+		type CurveParameterInput = FloatInput;
+		type CurveParameterType = Float;
+		type DefaultOrigin = EnsureSigned<AccountId>;
+		type DepositCurrency = Balances;
+		type DepositPerCurrency = CurrencyDeposit;
+		type ForceOrigin = EnsureRoot<AccountId>;
+		type Fungibles = Assets;
+		type MaxCurrencies = MaxCurrencies;
+		type MaxStringLength = StringLimit;
+		type PoolCreateOrigin = EnsureSigned<AccountId>;
+		type PoolId = AccountId;
+		type RuntimeEvent = RuntimeEvent;
+		type RuntimeHoldReason = RuntimeHoldReason;
+	}
+
+	#[derive(Clone, Default)]
+	pub(crate) struct ExtBuilder {
+		native_assets: Vec<(AccountId, DepositCurrencyBalanceOf<Test>)>,
+		bonded_balance: Vec<(AssetId, AccountId, Balance)>,
+		// denomination, pool_id, PoolDetails
+		pools: Vec<(u8, AccountId, PoolDetailsOf<Test>)>,
+		collaterals: Vec<AssetId>,
+	}
+
+	impl ExtBuilder {
+		pub(crate) fn with_native_balances(
+			mut self,
+			native_assets: Vec<(AccountId, DepositCurrencyBalanceOf<Test>)>,
+		) -> Self {
+			self.native_assets = native_assets;
+			self
+		}
+
+		pub(crate) fn with_pools(mut self, pools: Vec<(u8, AccountId, PoolDetailsOf<Test>)>) -> Self {
+			self.pools = pools;
+			self
+		}
+
+		pub(crate) fn with_bonded_balance(mut self, bonded_balance: Vec<(AssetId, AccountId, Balance)>) -> Self {
+			self.bonded_balance = bonded_balance;
+			self
+		}
+
+		pub(crate) fn build(mut self) -> sp_io::TestExternalities {
+			let mut storage = frame_system::GenesisConfig::<Test>::default().build_storage().unwrap();
+			pallet_balances::GenesisConfig::<Test> {
+				balances: self.native_assets.clone(),
+			}
+			.assimilate_storage(&mut storage)
+			.expect("assimilate should not fail");
+
+			let collateral_assets = self.collaterals.into_iter().map(|id| (id, ACCOUNT_99, false, 0));
+
+			pallet_assets::GenesisConfig::<Test> {
+				assets: self
+					.pools
+					.iter()
+					.map(|(_, owner, pool)| {
+						pool.bonded_currencies
+							.iter()
+							.map(|id| (*id, owner.to_owned(), false, 1u128))
+							.collect::<Vec<(AssetId, AccountId, bool, Balance)>>()
+					})
+					.flatten()
+					.chain(collateral_assets)
+					.collect(),
+
+				accounts: self.bonded_balance,
+				metadata: self
+					.pools
+					.iter()
+					.map(|(denomination, _, pool_details)| {
+						pool_details
```

Comment:
```suggestion
					.map(|( _, pool_details)| {
					    let denomination = pool_details.denomination.clone();
						pool_details
```

Reply:
is that the denomination we need?

Reply:
yes it is :raised_hands: 

I have update the implementation in: [e2ce89c](https://github.com/KILTprotocol/kilt-node/pull/780/commits/e2ce89cbb5fa2dac5efd5d8c5bf1dfee9568fee3)